### ISSUE TITLE
Review some security configurations for one click deployments

### ISSUE THREAD

**Issue by mcarrilloco** [hashedin/squealy#232]
Hi again,

After going trough the code a few times to make docker images and deploy inside our infrastructure I realized that this project might be being deployed with default security configurations that Django asks you to change before running in production. I'll bring them up for discussion on how to address them. The ones I have found are:

1) `DEBUG=True` Related source: https://docs.djangoproject.com/en/1.11/ref/settings/#debug

2) `SECRET_KEY` to hard coded value. Related: https://docs.djangoproject.com/en/1.11/ref/settings/#secret-key

3) `ALLOWED_HOSTS = ['*']` Related: https://docs.djangoproject.com/en/1.11/ref/settings/#allowed-hosts

I don't really understand Heroku one click deployments but I see some of the challenges to make the changes. I think using environmental variables and asking the users to change them when deploying is the way to go.

In our custom deployment I have been able to patch most of them, maintaining a parallel version of the code, except for the `DEBUG` one. When set to `False` the site starts to fail with this error `ValueError: The file 'css/bootstrap.min.css' could not be found with <whitenoise.storage.CompressedManifestStaticFilesStorage object at 0x7f49fb148bd0>.`

Hope this is useful. Let me know if something is not clear or there's something else I can do. Thanks!

**Comment by swapnil95** [comment]
We are using whitenoise to serve the static files in production. In order to serve the static files on production, you have to execute the following command during deployment:

```shell
python manage.py collectstatic
```

Let me know if you had already executed that command during the deployment and it didn't fix your issue

**Comment by mcarrilloco** [comment]
Hi @swapnil95. I just added the command to the docker file and the image can now be served with `DEBUG=False`. Thanks! The command solved this last issue for me. And I didn't know about `whitenoise`, certainly interesting.

Besides that, since I don't fully understand Heroku's pipeline I did a new one click deployment to verify that what I wrote here actually applied and there was not some hidden feature that would replace Django settings file. With this I was able to confirm that one click deployments run with the default values `DEBUG`, `SECRET_KEY` and `ALLOWED_HOSTS` that are specified here https://github.com/hashedin/squealy/blob/phase-2/squealyproj/settings.py#L27L33. There's no way to change these parameters without forking and maintaining a parallel version of the project. So what I wrote here still applies.

My main concern is that people who are running the one click heroku deployments on production and/or people who don't know that these values are insecure have no way of knowing that they might be deploying insecure applications. I think using environmental variables with secure defaults will be a simple yet effective to configure these parameters.

Thanks again for the clarification and you're fast and detailed response, certainly helped me finish our custom deployment.

**Comment by swapnil95** [comment]
Yes, you are right on the security part. The one click deployment should not run with default values of the variables you have mentioned. We will fix this
